src/engine/N3Base/N3UIList.cpp

```diff
@@ -195,6 +195,17 @@ bool CN3UIList::SetString(int iIndex, const std::string & szString) {
     return false;
 }
 
+CN3UIString * CN3UIList::GetChildStrFromList(std::string str) {
```

Comment:
To avoid memory copying, always pass strings by const reference.
```suggestion
CN3UIString * CN3UIList::GetChildStrFromList(const std::string & str) {
```